### ISSUE TITLE
Custom date ranges not working properly

### ISSUE THREAD

**Issue by pippinsplugins** [easydigitaldownloads/EDD-Commissions#82]
The custom date ranges in reports are not working properly. For example:

![screen shot 2015-08-13 at 4 55 10 pm](https://cloud.githubusercontent.com/assets/1034109/9262631/097b88d4-41dc-11e5-98f1-260bea17e5b4.png)

That should start on June 15 but is actually starting on June 1. It should end on July 15 but is ending on July 31.

**Comment by ghost1227** [comment]
I can only replicate the starting date bug. ending date seems to be working fine for me.

**Comment by ghost1227** [comment]
I'll admit to being a bit lost on this one. Given his recent work on the core reporting, might want to get @sunnyratilal involved here.

**Comment by ghost1227** [comment]
@sunnyratilal Have you had a chance to look at this?

**Comment by sunnyratilal** [comment]
PR opened at #193 that fixes the date ranges not working properly. There was a fair bit of code that wasn't respecting all the options we provide for reporting hence incorrectly leaving `$day_by_day` as `false`.

As much as I have tried, I cannot speed up the reporting to match the work I did with EDD because of the way we store the metadata.  As it is serialised, it has to be fetched and processed in PHP - I had one implementation that used hacky SQL but it wasn't ideal and wasn't have 100% reliable either - it's effectively relying on RegEx; not ideal when working with serialised data.

Again, same as what I said with #104, we really need to redefine the schema if we want performance improvements. I can replicate the performance of EDD only if the `amount` is stored in its own separate row in the `wp_postmeta` table.
